### PR TITLE
Add C670 test, anonymized test data, and update expected reports

### PR DESCRIPTION
**Pull request by edsu7**

### CODE REVIEW

File: seq_tools/validation/c670_rg_is_paired_in_bam.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            self.message = message
+            self.status = 'INVALID'
+            return
+    
+            
+        offending_rgs ={}
+        for rg in self.metadata.get('read_groups'):
+        
+            if 'is_paired_end' not in rg:
+                message = "'is_pair_end' not found in readgroup : %s" % rg['submitter_read_group_id']
```
Comment (lindaxiang) [inline seq_tools/validation/c670_rg_is_paired_in_bam.py:50]:
```
"Field 'is_paired_end' is not found for read_group  : %s in metadata JSON" % rg['submitter_read_group_id']
```

File: seq_tools/validation/c670_rg_is_paired_in_bam.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                self.logger.info(f'[{self.checker}] {message}')
+                self.message = message
+                self.status = 'INVALID'
+                return
+            elif rg['is_paired_end']==None:
+                message = "'is_pair_end' value in readgroup is null. Must be boolean: %s" % rg['submitter_read_group_id']
```
Comment (lindaxiang) [inline seq_tools/validation/c670_rg_is_paired_in_bam.py:56]:
```
" Field 'is_paired_end' for read_group: %s in metadata JSON is null. Must be boolean." % rg['submitter_read_group_id']
```

File: seq_tools/validation/c670_rg_is_paired_in_bam.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                stderr=subprocess.STDOUT,
+                shell=True
+            )
+            paired_check_bool = True if "10" in paired_check.decode('utf-8').rstrip().rstrip().split('\n') else False
+            paired_metadata_bool = rg['is_paired_end']
+            if paired_check_bool != paired_metadata_bool:
+                offending_rgs[rg['file_r1']].append(rg['submitter_read_group_id'])
+                
+                self.status = 'INVALID'
+                message = "Read group paired status in BAM does not match metadata: %s" % rg['submitter_read_group_id']
```
Comment (lindaxiang) [inline seq_tools/validation/c670_rg_is_paired_in_bam.py:79]:
```
"Paired-end status in BAM does not match field 'is_paired_end' in metadata JSON for read_group: %s" % rg['submitter_read_group_id']
```

File: seq_tools/validation/c670_rg_is_paired_in_bam.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    def check(self):
+        # status already set at initiation
+        if self.status:
+            return
+
+        if not self.metadata.get('read_groups'):
+            message = "Missing 'read_groups' section in the metadata JSON"
+            self.logger.info(f'[{self.checker}] {message}')
+            self.message = message
+            self.status = 'INVALID'
```
Comment (lindaxiang) [inline seq_tools/validation/c670_rg_is_paired_in_bam.py:42]: What will report in the validation report doc for this case?

File: seq_tools/validation/c670_rg_is_paired_in_bam.py (hunk 1 above)
Comment (lindaxiang) [reply seq_tools/validation/c670_rg_is_paired_in_bam.py:50]: same question, what will be reported for this use case?

File: seq_tools/validation/c670_rg_is_paired_in_bam.py (hunk 2 above)
Comment (lindaxiang) [reply seq_tools/validation/c670_rg_is_paired_in_bam.py:56]: same question, what will be reported for this use case?

File: seq_tools/validation/c670_rg_is_paired_in_bam.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+                self.message = message
+                self.logger.info(f'[{self.checker}] {message}')
+
+        
+        msg = []
+        for rg in offending_rgs.keys():
+            if len(offending_rgs[rg])>0:
+                msg.append("Offending read groups in BAM %s: %s" % (rg,",".join(offending_rgs[rg])))
+        if msg:    
+            message = "Paired status does not match in the following: %s" % ('; '.join(msg))
```
Comment (lindaxiang) [inline seq_tools/validation/c670_rg_is_paired_in_bam.py:89]:
```
"Paired-end status in BAM does not match field 'is_paired_end' in metadata JSON for the following read_groups: %s" % ('; '.join(msg))
```

**Review by lindaxiang — CHANGES_REQUESTED** [review]

File: seq_tools/validation/c650_sm_in_bam_matches_metadata.py [hunk 6]
```diff
@@ -91,7 +91,11 @@ def check(self):
         if bams and len(all_sms) != 1:
             raise Exception("No SM or more than one SM found in BAM(s): '%s'. Please see earlier check status "
                             "for details." % ', '.join(all_sms))
-
+        elif bams and list(all_sms)[0]=="":
+            self.status = 'WARNING'
+            message = "SM is empty. Validation status: WARNING. NOTE that submitterSampleId in metadata JSON will be used in the header of ARGO uniformly aligned sequences."
```
[the remark this answers is not in the document]
Comment (edsu7) [reply seq_tools/validation/c650_sm_in_bam_matches_metadata.py]: Updated message
https://github.com/icgc-argo/seq-tools/blob/55a34779f106549fd3809cc495c4755169174db5/seq_tools/validation/c650_sm_in_bam_matches_metadata.py#L96

File: seq_tools/validation/c670_rg_is_paired_in_bam.py (hunk 2 above)
Comment (edsu7) [reply seq_tools/validation/c670_rg_is_paired_in_bam.py]: Added test case to demonstrate report
https://github.com/icgc-argo/seq-tools/blob/55a34779f106549fd3809cc495c4755169174db5/tests/expected_reports/anon_chr1_rnaseq/anon_chr1_meta_ispaired_null.validation_report.INVALID.jsonl#L22

File: seq_tools/validation/c670_rg_is_paired_in_bam.py (hunk 3 above)
Comment (edsu7) [reply seq_tools/validation/c670_rg_is_paired_in_bam.py]: Updated message
https://github.com/icgc-argo/seq-tools/blob/55a34779f106549fd3809cc495c4755169174db5/seq_tools/validation/c670_rg_is_paired_in_bam.py#L79

File: seq_tools/validation/c670_rg_is_paired_in_bam.py (hunk 4 above)
Comment (edsu7) [reply seq_tools/validation/c670_rg_is_paired_in_bam.py:42]: Added test case to demonstrate report
https://github.com/icgc-argo/seq-tools/blob/55a34779f106549fd3809cc495c4755169174db5/tests/expected_reports/anon_chr1_rnaseq/anon_chr1_meta_no_rgs.validation_report.INVALID.jsonl#L52

File: seq_tools/validation/c670_rg_is_paired_in_bam.py (hunk 5 above)
Comment (edsu7) [reply seq_tools/validation/c670_rg_is_paired_in_bam.py]: Added message
https://github.com/icgc-argo/seq-tools/blob/55a34779f106549fd3809cc495c4755169174db5/seq_tools/validation/c670_rg_is_paired_in_bam.py#L89

**Review by lindaxiang — APPROVED** [review]

**Pull request merged**